test.js

```diff
@@ -158,6 +154,27 @@ describe('capture-exit', function() {
         exit.onExit(function () { });
       }).to.throw('Cannot install handler when exit is not captured.  Call `captureExit()` first');
     });
+
+    it('throws if an exit is already happening', function() {
```

Comment:
@stefanpenner - Yes, we are self-imposing this error (adding an additional exit handler via `captureExit.onExit` during the exit process) because it is not trivial to reason about and we can always add support back if there is a good use case presented.  I don't feel _super_ strongly that we must throw, but the specific semantics about how things would work is non-obvious.  

Try running these scripts (with the branch here):

```js
var captureExit = require('capture-exit');
captureExit.captureExit();

process.on('exit', function() {
  console.log('`process.on("exit"` firing');

  captureExit.onExit(function() {
    console.log('captureExit.onExit firing');
  });
});
console.log('exiting naturally');
```

```js
var captureExit = require('capture-exit');
captureExit.captureExit();

captureExit.onExit(function() {
  console.log('first `captureExit.onExit` firing');

  captureExit.onExit(function() {
    console.log('second `captureExit.onExit` firing');
  });
});
console.log('exiting naturally');
```

Note that in the first case, our `captureExit.onExit` handler **never** executes (this is due to node's constraint about `process.on('exit'` handlers being completely sync).

---

I think the *tldr;* summary (from my perspective) here is that this library's entire purpose is to ensure that when folks call `captureExit.onExit(function() { })` they are **guaranteed** to be ran (this is certainly ember-cli's use case, where we **must** ensure `tmp/` is cleaned).  We simply cannot guarantee that exit handlers will be run in the case of the first script in this comment.